rows.go

```diff
@@ -180,12 +238,111 @@ type HugeInt struct {
 	upper int64
 }
 
+func (v HugeInt) UUID() []byte {
+	var uuid [16]byte
+	// We need to flip the `sign bit` of the signed `HugeInt` to transform it to UUID bytes
+	binary.BigEndian.PutUint64(uuid[:8], uint64(v.upper)^1<<63)
+	binary.BigEndian.PutUint64(uuid[8:], v.lower)
+	return uuid[:]
+}
+
 func (v HugeInt) Int64() (int64, error) {
 	if v.upper == 0 && v.lower <= math.MaxInt64 {
 		return int64(v.lower), nil
-	} else if v.upper == -1 && v.lower <= math.MaxUint64 {
+	} else if v.upper == -1 {
 		return -int64(math.MaxUint64 - v.lower + 1), nil
 	} else {
 		return 0, fmt.Errorf("can not convert duckdb:hugeint to go:int64 (upper:%d,lower:%d)", v.upper, v.lower)
 	}
 }
+
+func scanList(vector C.duckdb_vector, rowIdx C.idx_t) ([]any, error) {
+	data := C.duckdb_list_vector_get_child(vector)
+	entry := get[duckdb_list_entry_t](vector, rowIdx)
+	converted := make([]any, 0, entry.length)
```

Comment:
Yeah, this is an interesting one. Your solution is definitely also correct. 

Not sure if this applies in go, but one reason I often use the `append` solution is you don't have fill the entire slice with zero values in `make` just to write over it immediately.
In terms of performance within the loop, I would say it's probably about the same as just setting the index because we never have to resize due to the preallocation.

I think go's alloc is not defined in `go`, but if we have a look at rust's equivalent function `Vec::push`. I feel pretty confident in saying it's not a performance issue.

```
let end = self.as_mut_ptr().add(self.len);
ptr::write(end, value);
self.len += 1;
```

Having said all that, happy to change it if you prefer the other style!

Reply:
I'm totally fine with your solution. Just did a bit of googling and [someone has benchmarked](https://stackoverflow.com/questions/38654729/golang-slice-append-vs-assign-performance) it, but it is nearly the same. I think `i - entry.offset` as index is less readable as `append` and not worth the few ns difference.